index.js

```diff
@@ -462,9 +463,17 @@ class Squeakquel extends Datastore {
 
             // every other selected field must be aggregated so database engine won't complain
             // use "MAX" since the nature of this table is append-only
-            findParams.attributes = includedFields.map(field =>
-                [Sequelize.fn('MAX', Sequelize.col(field)), field]
-            );
+            findParams.attributes = includedFields.map((field) => {
+                let col = Sequelize.col(field);
+
+                // Cast boolean to integer
+                // It is safer for most dialect to cast to integer instead of other integer type like smallint
+                if (fields[field] && fields[field].type === 'boolean') {
+                    col = Sequelize.cast(col, 'integer');
```

Comment:
if the filed has empty value, when cast into an integer, what is the value going to be?

Reply:
Empty again

Reply:
or should i say NULL got ignored so it wont matter in aggregation?